aws/resource_aws_ssm_maintenance_window_task.go

```diff
@@ -31,19 +31,22 @@ func resourceAwsSsmMaintenanceWindowTask() *schema.Resource {
 			},
 
 			"max_concurrency": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^([1-9][0-9]*|[1-9][0-9]%|[1-9]%|100%)$`), ""),
```

Comment:
`validation.StringMatch()` errors should include a human friendly message over the raw regular expression. Reference: https://github.com/hashicorp/terraform-provider-aws/issues/16628

```suggestion
				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^([1-9][0-9]*|[1-9][0-9]%|[1-9]%|100%)$`), "must be a number without leading zeros or a percentage between 1% and 100% without leading zeros and ending with the percentage symbol"),
```